Helm Chart Install Reference (#1241)
spectrocloud/librarium: content/docs/12-enterprise-version/03.5-helm-chart-install-reference.md

Reviewer: The default values column is inconsistent. Some start as capitalized, others use code tickets. We can't have all these inconsistencies. Also, some of the default values with `""` have double quotes that are curved. This comes from a copy and paste action. Let's fix these double quotes to look correct.

Author: Fixed it.

icon: ""
hideToC: False
fullWidth: False
---

# Helm Chart Install Reference 

You can use the Palette Helm Chart to install Palette in a multi-node Kubernetes cluster in your production environment. The Helm chart allows you to customize values in the values.yaml file. This reference guide lists and describes parameters available in the values.yaml file from the Helm Chart for your installation. To learn how to install Palette using the Helm Chart, refer to [Helm Chart Mode](https://docs.spectrocloud.com/enterprise-version/deploying-palette-with-helm).  

# Required parameters

The following parameters are required:

- **env.rootDomain**- Used to configure the domain for the Palette installation. **Note:** You should create a DNS record separately, and it should be a wildcard to account for Organization prefixes. Review the [environment parameters](#environment) to learn more. 

- **natsUrl**- The URL format specifies configuring the NATS server(s) to the IP address and port. Review the [Network Address Translation (NATS) parameters](#network-address-translationnats) to learn more. 
 

> - Refer to the [Registry and Palette Artifact Repository parameters](#registry-and-palette-artifact-repositorypar) used to specify the Docker registry where the chart images are stored and the Spectro Cloud Admission Registrar (SCAR), respectively.

Reviewer: This paragraph should start with the name of the parameter or section. It's a bit awkward in it's current form. Also, I noticed you are using PAR in the doc, so we should remove SCAR and replace with PAR.

| `saml.apiVersion` | `v1` | String |**SSO SAML** API version to use.| Optional |

```yaml
config:
  sso:
    saml:
      enabled: false
      acsUrlRoot: "myfirstpalette.spectrocloud.com"
      acsUrlScheme: "https"
      audienceUrl: "https://www.spectrocloud.com"
      entityId: "https://www.spectrocloud.com"
      apiVersion: "v1"
```

## Email

The following table provides a reference for the available **values.yaml** parameters for configuring email settings in Palette's self-hosted mode. 

| **Parameters** | **Default value** | **Type** | **Description** | **Required/Optional** |
| --- | --- | --- | --- | --- |
| `enabled` | `False` | Boolean | Specifies whether to enable email configuration. | Optional|
| `emailID ` | "" | String | Email address for sending mail.| Optional|
| `smtpServer` | `smtp.gmail.com` | String | SMTP server used for sending mail. | Optional |
| `smtpPort` | `587` | Integer | SMTP port used for sending mail.| Optional|
| `insecureSkipVerifyTIs` | `True` | Boolean | Specifies whether to skip TLS verification for the SMTP connection.| Optional |
| `fromEmailID` | `noreply@spectrocloud.com` | String |Email address used from From address.| Optional |

> | `password` | `YmFzZTY0X2VtYWlsX3Bhc3N3b3Jk` | String |Base64-encoded SMTP password when sending emails.| Optional |

Reviewer: ```suggestion
| `password` | `""` | String |Base64-encoded SMTP password when sending emails.| Optional |
```
We don't hardcode an email password as a default value.

---
title: "Helm Chart Install Reference"
metaTitle: "Helm Chart Install References"
metaDescription: "Reference page for Pallete Helm chart self-installation values.yaml file parameters."
icon: ""
hideToC: False
fullWidth: False
---

# Helm Chart Install Reference 

You can use the Palette Helm Chart to install Palette in a multi-node Kubernetes cluster in your production environment. The Helm chart allows you to customize values in the values.yaml file. This reference guide lists and describes parameters available in the values.yaml file from the Helm Chart for your installation. To learn how to install Palette using the Helm Chart, refer to [Helm Chart Mode](https://docs.spectrocloud.com/enterprise-version/deploying-palette-with-helm).  

> # Required parameters

Reviewer: ```suggestion
# Required Parameters
```
Capitalize all the first letters, title case.

The following parameters are required:

> - **env.rootDomain**- Used to configure the domain for the Palette installation. **Note:** You should create a DNS record separately, and it should be a wildcard to account for Organization prefixes. Review the [Environment parameters](#environment) to learn more. 

Reviewer: ```suggestion
- **env.rootDomain**- Used to configure the domain for the Palette installation.  You should create a CNAME DNS record separately, and it should be a wildcard to account for Organization prefixes. Review the [Environment parameters](#environment) to learn more. 
```

Let's remove Note:, just add it into the description otherwise it looks odd.

> - **natsUrl**- The URL format specifies configuring the NATS server(s) to the IP address and port. Review the [Network Address Translation (NATS) parameters](#network-address-translationnats) to learn more. 

Reviewer: ```suggestion
- **natsUrl**- The URL format specifies configuring the NATS servers to the IP address and port. Review the [Network Address Translation (NATS) parameters](#network-address-translationnats) to learn more. 
```
Default to plural

> - **Registry and Palette Artifact Repository parameters**- Refer to the [Registry and Palette Artifact Repository parameters](#registry-and-palette-artifact-repositorypar) used to specify the Docker registry where the chart images are stored and the **Palette Artifact Repository(PAR)**, respectively.

Reviewer: ```suggestion
- **Registry and Palette Artifact Repository parameters**- Refer to the [Registry and Palette Artifact Repository parameters](#registry-and-palette-artifact-repositorypar) used to specify the Docker registry where the chart images are stored and the **Palette Artifact Repository (PAR)**, respectively.
```
Space missing

# MongoDB 

Palette uses MongoDB as its database and supports two modes of deployment: 

> 1. MongoDB running inside the cluster. 

Reviewer: ```suggestion
1. MongoDB deployed and active inside the cluster. 
```
Avoid ablism words.

| **Parameters** | **Default value** | **Type** | **Description** | **Required/Optional** |
| --- | --- | --- | --- | --- |
| `env.rootDomain` | `""` | String | Specifies the default DNS record mapped to the ingress-nginx-controller load balancer. It is mandatory if false; otherwise, leave it empty. | Required|
| `env.installerMode` | `self-hosted` | String | Specifies the installer mode. Do not modify the value.| Required|
| `env.installerCloud` | `""` | String | Specifies cloud provider. It shoudl be left empty. | Optional |

```yaml
config:
    env:
    rootDomain: ""
    installerMode: "self-hosted"
    installerCloud: ""
```

## Cluster 

Cluster parameters are used to specify the Kubernetes cluster configuration.

| **Parameters** | **Default value** | **Type** | **Description** | **Required/Optional** |
| --- | --- | --- | --- | --- |
| `stableEndpointAccess` | `False` | Boolean | Specifies whether EKS cluster is deployed in private network access through a stable private endpoint, if true, through a public endpoint. | Optional|

```yaml
config:
    cluster:
    stableEndpointAccess: false
```

> ## Registry and Palette Artifact Repository(PAR) 

Reviewer: ```suggestion
## Registry and Palette Artifact Repository (PAR) 
```
Missing space.

| `registry.username` | String | The username for the registry.                           | Required           |
| `scar.endpoint`     | String | The endpoint URL of the **Palette Artifact Repository(PAR)**. | Required           |
| `scar.username`     | String | The username for the **PAR**.                               | Required           |
| `scar.password`     | String | The password for the **PAR**.                               | Required           |

```yaml
config:
    registry:
    endpoint: "<Contact Spectro Cloud Sales for More info>"
    name: "<Contact Spectro Cloud Sales for More info>"
    password: "<Contact Spectro Cloud Sales for More info>"
    username: "<Contact Spectro Cloud Sales for More info>"

    scar:
    endpoint: "<Contact Spectro Cloud Sales for More info>"
    username: "<Contact Spectro Cloud Sales for More info>"
    password: "<Contact Spectro Cloud Sales for More info>"
```

Please contact the Spectro Cloud support team at [support@spectrocloud.com](support@spectrocloud.com) to receive the required values and credentials for **Palette Artifact Repository(PAR**).

> # Network Address Translation(NATS) 

Reviewer: ```suggestion
# Network Address Translation (NATS) 
```
Missing space.

| `natsUrl`    |     `""`    | String  | Comma-separated list of <dns_name:port> mappings for nats load balancer service. Mandatory if `nats.internal` is false.     | Required           |
| `annotations`| `{}`          | Map     | A map of key-value pairs that specify the load balancer annotations for NATS. These annotations vary depending on the cloud provider.                                                                                           | Optional           |
| `routes`     | `[]`          | List    | List of server URLs for clustering (excluding self-routes) that can include authentication via token or username/password in the URL.                                               | Optional           |
| `natsStaticIP`| `""`         | String  | Static IP for the NATS load balancer service. If empty, a dynamic IP will be generated.                                   | Optional           |

```yaml
nats:
  internal: true
  natsUrl: ""
  annotations: {}
  routes: []
  natsStaticIP: ""
```

# Ingress 

> The **Ingress** block in the YAML configuration file is used for configuring the **NGINX Ingress** Controller, which provides an external HTTP(S) load balancer for **Kubernetes** services. 

Reviewer: ```suggestion
The **Ingress** block in the YAML configuration file is used for configuring the **NGINX Ingress** Controller, which provides an external HTTP load balancer for **Kubernetes** services. 
```

| `key`                          | `""`          | String  | Default SSL key for the NGINX Ingress Controller.                                                                                        | Optional           |
| `annotations`                  | `{}`          | Map     | A map of key-value pairs that specify the load balancer annotations for Ingress.                                                                      | Optional           |
| `ingressStaticIP`              | `""`          | String  | Static IP for the Ingress load balancer service. If empty, a dynamic IP will be generated.                                                        | Optional           |
| `terminateHTTPSAtLoadBalancer` | `False`       | Boolean | Specify whether to terminate HTTPS at the load balancer.                 | Optional           |

```yaml
ingress:
  ingress:
    internal: false
    certificate: ""
    key: ""
    annotations: {}
    ingressStaticIP: ""
    terminateHTTPSAtLoadBalancer: false
```

# Spectro Proxy

> “**Spectro Proxy**” is a section of the configuration file for the Spectro server-side proxy.

Reviewer: ```suggestion
**Spectro Proxy** is a section of the configuration file for the Spectro server-side proxy.
```
This should not be in quotes.

---
title: "Helm Chart Install Reference"
metaTitle: "Helm Chart Install References"
metaDescription: "Reference page for Pallete Helm chart self-installation values.yaml file parameters."
icon: ""
hideToC: False
fullWidth: False
---

# Helm Chart Install Reference 

You can use the Palette Helm Chart to install Palette in a multi-node Kubernetes cluster in your production environment. The Helm chart allows you to customize values in the values.yaml file. This reference guide lists and describes parameters available in the values.yaml file from the Helm Chart for your installation. To learn how to install Palette using the Helm Chart, refer to [Helm Chart Mode](/enterprise-version/deploying-palette-with-helm).

# Required Parameters

The following parameters are required:

> 

Author: Hi Rita, Thanks for this.

I added the break. It looks cleaner now.

- **env.rootDomain** - Used to configure the domain for the Palette installation. You should create a CNAME DNS record separately, and it should be a wildcard to account for Organization prefixes. Review the [Environment parameters](/enterprise-version/helm-chart-install-reference#environment) to learn more. <br /> <br />

- **natsUrl** - The URL format specifies configuring the NATS servers to the IP address and port. Review the [Network Address Translation (NATS) parameters](/enterprise-version/helm-chart-install-reference#networkaddresstranslation(nats)) to learn more. <br /> <br />

 
 
- **Registry and Palette Artifact Repository** - Refer to the [Registry and Palette Artifact Repository parameters](/enterprise-version/helm-chart-install-reference#registryandpaletteartifactrepository(par)) used to specify the Docker registry where the chart images are stored and the Palette Artifact Repository (PAR), respectively.

# MongoDB 

> 

Author: Yep. I did it exactly how you asked it here, it seems concise enough.
